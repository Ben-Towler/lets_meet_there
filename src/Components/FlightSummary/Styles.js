import styled from 'styled-components';

export const StyledFlightSummary = styled.div`
  background-color: green;
`