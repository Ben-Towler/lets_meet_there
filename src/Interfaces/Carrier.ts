export interface Carrier {
  CarrierId: number;
  Name: String;
}