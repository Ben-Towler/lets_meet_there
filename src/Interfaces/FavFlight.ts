interface FavLeg {
  CarrierIds: number[]; 
  OriginId: number;
  DestinationId: number; 
  DepartureDate: string;
}


export interface FavFlight {
  Direct: Boolean;
  InboundLeg: FavLeg;
  MinPrice: number;
  OutboundLeg: FavLeg;
  QuoteDateTime: string;
  QuoteId: number;
}