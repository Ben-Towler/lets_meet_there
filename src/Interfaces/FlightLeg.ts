export interface FlightLeg {
  CarrierIds: number[],
  OriginId: number,
  DestinationId: number,
  DepartureDate: string
}