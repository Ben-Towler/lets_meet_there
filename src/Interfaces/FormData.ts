export interface FormData {
  location?: string;
  destination?: string;
  startDate?: string;
  endDate?: string;
  name?: string;
}