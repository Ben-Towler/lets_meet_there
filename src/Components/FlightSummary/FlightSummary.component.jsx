import React from 'react';
import { StyledFlightSummary } from './Styles.js';

function FlightSummary ({ location }) {

  return (
    <StyledFlightSummary>
      {location.city}
      {location.country}
    </StyledFlightSummary>
  );
  
};

export default FlightSummary;
