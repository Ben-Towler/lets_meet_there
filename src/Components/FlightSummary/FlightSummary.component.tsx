import React from 'react';
import { StyledFlightSummary } from './Styles';

interface Props {
  location: any
}

function FlightSummary ({ location }: Props): JSX.Element {

  return (
    <StyledFlightSummary>
      {location.city}
      {location.country}
    </StyledFlightSummary>
  );
  
}

export default FlightSummary;
