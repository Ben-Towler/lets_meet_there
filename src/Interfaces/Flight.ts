import {FlightLeg} from './FlightLeg';

export interface Flight {
  QuoteId: number;
  MinPrice: number;
  Direct: Boolean,
  OutboundLeg: FlightLeg,
  InboundLeg: FlightLeg,
  QuoteDateTime: string,
}



