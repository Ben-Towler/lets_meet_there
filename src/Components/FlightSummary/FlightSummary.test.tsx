import React from 'react';
import renderer from 'react-test-renderer';
import FlightSummary from './FlightSummary.component';
import {Location} from 'Interfaces/Location';
 
describe('Flight Summary', () => {
  let location: Location;

  beforeEach(() => {
    location = {
      city: 'Barcelona',
      country: 'Spain'
    }
  });

  test('snapshot renders', () => {
    const component = renderer.create(
    <FlightSummary
      location={location}
    />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});