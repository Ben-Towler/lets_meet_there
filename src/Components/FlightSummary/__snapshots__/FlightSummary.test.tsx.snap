// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight Summary snapshot renders 1`] = `
<div
  className="sc-AxjAm fDEmJb"
>
  Barcelona
</div>
`;
